File: frontEnd/src/core/services/workshop.service.ts
import { WorkShop } from './../model/workshop.model';
import { BaseService, IBaseService } from './base.service';
export interface IWorkShopService extends IBaseService<WorkShop> {}

export class WorkShopService extends BaseService<WorkShop> implements IWorkShopService {

    constructor(controller: string) {
        super(controller)
    }
   
}

File: backEnd/GenericApi.Services/Services/WorkShopService.cs
using AutoMapper;
using FluentValidation;
using GenericApi.Bl.Dto;
using GenericApi.Model.Entities;
using GenericApi.Model.Repositories;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GenericApi.Services.Services
{ 
    public interface IWorkShopService : IBaseService<WorkShop, WorkShopDto> {}
    public class WorkShopService : BaseService<WorkShop, WorkShopDto>, IWorkShopService
    {
        public WorkShopService(
            IWorkShopRepository repository, 
            IMapper mapper, 
            IValidator<WorkShopDto> validator) : base(repository,mapper, validator)
        {
        }
    }
}
